packages/datadog_session_replay/android/src/test/kotlin/com/datadoghq/flutter/sessionreplay/resource/ResourceResolverTest.kt

```diff
@@ -0,0 +1,263 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2025-Present Datadog, Inc.
+ */
+
+package com.datadoghq.flutter.sessionreplay.resource
+
+import android.graphics.Bitmap
+import assertk.assertThat
+import assertk.assertions.isEqualTo
+import assertk.assertions.isNotEqualTo
+import assertk.assertions.isNull
+import com.datadog.android.api.InternalLogger
+import fr.xgouchet.elmyr.annotation.IntForgery
+import fr.xgouchet.elmyr.annotation.StringForgery
+import fr.xgouchet.elmyr.junit5.ForgeExtension
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.verify
+import java.nio.ByteBuffer
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+
+@ExtendWith(ForgeExtension::class)
+internal class ResourceResolverTest {
+    val mockInternalLogger = mockk<InternalLogger>()
+    val mockResourcesWriter = mockk<ResourceWriter>()
+    val mockBitmapHandler = mockk<BitmapHandler>()
+
+    fun createFakeImage(width: Int = 25, height: Int = 25, filledWith: Byte = 0): ByteBuffer {
```

Comment:
minor: should we randomize the image dimensions? Same question regarding all the usages of 25 here